File: backend/app/schemas/channel.py
"""
渠道相关的 Pydantic Schema
"""
from pydantic import BaseModel, Field, field_validator
from typing import Optional, Dict, Any
from datetime import datetime


class ChannelBase(BaseModel):
    """渠道基础 Schema"""
    name: str = Field(..., min_length=1, max_length=100, description="渠道名称")
    channel_type: str = Field(..., description="渠道类型（ecommerce/notification）")
    platform: str = Field(..., min_length=1, max_length=50, description="平台标识")
    description: Optional[str] = Field(None, description="渠道描述")
    config: Dict[str, Any] = Field(..., description="渠道配置")
    is_default: bool = Field(False, description="是否为默认渠道")

    @field_validator('channel_type')
    @classmethod
    def validate_channel_type(cls, v):
        if v not in ['ecommerce', 'notification']:
            raise ValueError('channel_type 必须是 ecommerce 或 notification')
        return v


class ChannelCreate(ChannelBase):
    """创建渠道 Schema"""
    pass


class ChannelUpdate(BaseModel):
    """更新渠道 Schema"""
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = None
    config: Optional[Dict[str, Any]] = None
    status: Optional[str] = Field(None, description="渠道状态（active/inactive/error）")
    is_default: Optional[bool] = None

    @field_validator('status')
    @classmethod
    def validate_status(cls, v):
        if v is not None and v not in ['active', 'inactive', 'error']:
            raise ValueError('status 必须是 active、inactive 或 error')
        return v


class ChannelResponse(ChannelBase):
    """渠道响应 Schema"""
    id: int
    status: str
    usage_count: int
    last_used_at: Optional[str]
    last_error: Optional[str]
    created_by: int
    created_at: str
    updated_at: str

    class Config:
        from_attributes = True


class ChannelTestRequest(BaseModel):
    """测试渠道连接请求"""
    channel_id: int = Field(..., description="渠道ID")


class ChannelTestResponse(BaseModel):
    """测试渠道连接响应"""
    success: bool = Field(..., description="测试是否成功")
    message: str = Field(..., description="测试结果消息")
    details: Optional[Dict[str, Any]] = Field(None, description="详细信息")


class ChannelStatistics(BaseModel):
    """渠道统计信息"""
    total_channels: int = Field(..., description="总渠道数")
    active_channels: int = Field(..., description="启用的渠道数")
    ecommerce_channels: int = Field(..., description="电商平台渠道数")
    notification_channels: int = Field(..., description="通知渠道数")
    total_usage: int = Field(..., description="总使用次数")

File: frontend/src/api/channel.ts
/**
 * 渠道管理 API 接口
 */
import { apiClient } from '../utils/request';

// 渠道信息接口
export interface Channel {
  id: number;
  name: string;
  channel_type: 'ecommerce' | 'notification';
  platform: string;
  description?: string;
  config: Record<string, any>;
  status: 'active' | 'inactive' | 'error';
  is_default: boolean;
  usage_count: number;
  last_used_at?: string;
  last_error?: string;
  created_by: number;
  created_at: string;
  updated_at: string;
}

// 创建渠道请求
export interface ChannelCreateRequest {
  name: string;
  channel_type: 'ecommerce' | 'notification';
  platform: string;
  description?: string;
  config: Record<string, any>;
  is_default?: boolean;
}

// 更新渠道请求
export interface ChannelUpdateRequest {
  name?: string;
  description?: string;
  config?: Record<string, any>;
  status?: 'active' | 'inactive' | 'error';
  is_default?: boolean;
}

// 测试渠道请求
export interface ChannelTestRequest {
  channel_id: number;
}

// 测试渠道响应
export interface ChannelTestResponse {
  success: boolean;
  message: string;
  details?: Record<string, any>;
}

// 渠道统计
export interface ChannelStatistics {
  total_channels: number;
  active_channels: number;
  ecommerce_channels: number;
  notification_channels: number;
  total_usage: number;
}

/**
 * 获取渠道列表
 */
export const getChannelList = (params?: {
  channel_type?: string;
  status?: string;
}) => {
  return apiClient.get<Channel[]>('/api/v1/channels/list', { params });
};

/**
 * 获取渠道详情
 */
export const getChannel = (channelId: number) => {
  return apiClient.get<Channel>(`/api/v1/channels/${channelId}`);
};

/**
 * 创建渠道
 */
export const createChannel = (data: ChannelCreateRequest) => {
  return apiClient.post<Channel>('/api/v1/channels/create', data);
};

/**
 * 更新渠道
 */
export const updateChannel = (channelId: number, data: ChannelUpdateRequest) => {
  return apiClient.put<Channel>(`/api/v1/channels/${channelId}`, data);
};

/**
 * 删除渠道
 */
export const deleteChannel = (channelId: number) => {
  return apiClient.delete<null>(`/api/v1/channels/${channelId}`);
};

/**
 * 测试渠道连接
 */
export const testChannel = (data: ChannelTestRequest) => {
  return apiClient.post<ChannelTestResponse>('/api/v1/channels/test', data);
};

/**
 * 获取渠道统计
 */
export const getChannelStatistics = () => {
  return apiClient.get<ChannelStatistics>('/api/v1/channels/statistics/summary');
};
